addons/purchase_requisition/tests/test_purchase_requisition.py

```diff
@@ -485,5 +469,40 @@ def test_08_purchase_requisition_sequence(self):
             'company_id': new_company.id,
         })
         self.bo_requisition.company_id = new_company
+        self.bo_requisition.line_ids.write({'price_unit': 10.0})
         self.bo_requisition.action_in_progress()
         self.assertTrue(self.bo_requisition.name.startswith("REQ_"))
+
+    def test_09_purchase_requisition_purchase_template(self):
+        quantity = 2
+        self.res_partner_2 = self.env['res.partner'].create({
+            'name': 'Azure Interior',
+        })
+
+        self.supplierinfo10 = self.env['product.supplierinfo'].create({
+            'partner_id': self.res_partner_2.id,
+            'product_id': self.product_09.id,
+            'price': 15.0,
+            'min_qty': quantity,
+        })
+
+        # Create a pruchase requisition with type purchase template and two product
+        line1 = (0, 0, {'product_id': self.product_09.id, 'product_uom_id': self.product_uom_id.id})
+        line2 = (0, 0, {'product_id': self.product_13.id, 'product_uom_id': self.product_uom_id.id})
+
+        requisition_purchase_template = self.env['purchase.requisition'].create({
+            'line_ids': [line1, line2],
+            'requisition_type': 'purchase_template',
+            'vendor_id': self.res_partner_2.id,
+        })
+
+        # update the product_qty to get the Unit price
+        requisition_purchase_template.line_ids[0].write({'product_qty': 2.0})
+        requisition_purchase_template.line_ids[1].write({'product_qty': 1.0})
+
+        # confirm the requisition
+        requisition_purchase_template.action_in_progress()
+
+        # Check for both product price_unit, fetch from supplierinfo if present otherwise it is standard_price
+        self.assertEqual(requisition_purchase_template.line_ids[0].price_unit, self.supplierinfo10.price, 'Unit Price should fetch from supplierinfo')
+        self.assertEqual(requisition_purchase_template.line_ids[1].price_unit, self.product_13.standard_price, 'Unit Price should standard_price when supplierinfo is not present')
```

Comment:
```suggestion
        self.assertEqual(purchase_template.line_ids[0].price_unit, self.supplierinfo10.price, 'Unit Price should match supplierinfo price')
        self.assertEqual(purchase_template.line_ids[1].price_unit, self.product_13.standard_price, 'Unit Price should equal standard_price when no supplierinfo')
```

Additionally, after this you should also create a Purchase Order using this template to ensure the values (quantity, unit price) match